lib/docker.js

```diff
@@ -0,0 +1,518 @@
+#!/usr/bin/env node
+
+
+'use strict';
+
+var spawn = require('child_process').spawn;
+var yaml = require('js-yaml');
+var P = require('bluebird');
+var fs = P.promisifyAll(require('fs'));
+var path = require('path');
+
+
+// info from the package definition
+var pkg;
+// info from the service-runner config file
+var config;
+// target info
+var targets;
+
+// the options used in the script
+var opts;
+// the image name
+var img_name;
+// the container's name
+var name;
+
+// holds the curently running process
+var child;
+
+
+/**
+ * Wraps a child process spawn in a promise which resolves
+ * when the child process exists.
+ *
+ * @param {Array} args the command and its arguments to run (uses /usr/bin/env)
+ * @param {Boolean} capture whether to capture stdout and return its contents
+ * @return {Promise} the promise which is fulfilled once the child exists
+ */
+function promised_spawn(args, capture) {
```

Comment:
For some executions I want to capture the output altogether (like `git status`), while for others it is good for the user to see the output in real time (like building the image, or `npm install`).